hls-fix.patch

```diff
@@ -0,0 +1,59 @@
+diff --git a/include/ap_int_base.h b/include/ap_int_base.h
+index 091552a..1d30fea 100644
+--- a/include/ap_int_base.h
++++ b/include/ap_int_base.h
+@@ -122,7 +122,7 @@ struct ap_int_base : public _AP_ROOT_TYPE<_AP_W, _AP_S> {
+    */
+   typedef typename retval<AP_MAX((_AP_W + 7) / 8, 8), _AP_S>::Type RetType;
+ 
+-  static const int width = _AP_W;
++  constexpr static const int width = _AP_W;
+ 
+   template <int _AP_W2, bool _AP_S2>
+   struct RType {
+diff --git a/include/etc/ap_private.h b/include/etc/ap_private.h
+index 0c29a0a..1684b13 100644
+--- a/include/etc/ap_private.h
++++ b/include/etc/ap_private.h
+@@ -1590,7 +1590,7 @@ ASSIGN_OP_FROM_INT(double)
+   }
+ 
+  public:
+-  INLINE ap_private() {
++  INLINE ap_private(): VAL(0) {
+     set_canary();
+     clearUnusedBits();
+     check_canary();
+@@ -3022,8 +3022,9 @@ class ap_private<_AP_W, _AP_S, false> {
+ #pragma warning(disable : 4521 4522)
+ #endif
+  public:
+-  enum { BitWidth = _AP_W, _AP_N = (_AP_W + 63) / 64 };
+-  static const int width = _AP_W;
++  static constexpr uint32_t BitWidth = _AP_W;
++  static constexpr uint32_t _AP_N = (_AP_W + 63) / 64;
++  constexpr static const int width = _AP_W;
```

Comment:
@makortel , converting `enum` to `static constexpr` fixed the runtime cling warnings like https://github.com/cms-sw/cmsdist/pull/9311#issue-2424355867 . Does this change look reasonable to you?

Reply:
I think `const` on L35 is reduntant, since constexpr is already const

Reply:
Converting to `static constexpr` is correct, and @iarspider is also right that the `const` is redundant here.